pkg/workflows/wasm/host/module.go

```diff
@@ -445,3 +469,46 @@ func fetchFn(logger logger.Logger, modCfg *ModuleConfig) func(caller *wasmtime.C
 		return ErrnoSuccess
 	}
 }
+
+func createEmitFn(
+	l logger.Logger,
+	e sdk.Emitter,
+) func(caller *wasmtime.Caller, msgptr, msglen int32) int32 {
+	logErr := func(err error) {
+		l.Errorf("error emitting message: %s", err)
+	}
+
+	return func(caller *wasmtime.Caller, msgptr, msglen int32) int32 {
+		b, err := safeMem(caller, msgptr, msglen)
+		if err != nil {
+			logErr(err)
+			return ErrnoFault
```

Comment:
The API can return an error, so can we return it in a response too? That would imply that we should have a resp buffer.

Reply:
this is a failure to read the binary's memory, added a response buffer for the error on calling Emit